src/compiler/scala/tools/nsc/transform/patmat/MatchTreeMaking.scala

```diff
@@ -650,6 +650,28 @@ trait MatchTreeMaking extends MatchCodeGen with Debugging {
       }
     }
 
+    private object Switchable {
+      val switchableTpe = Set(ByteTpe, ShortTpe, IntTpe, CharTpe, StringTpe)
```

Comment:
LGTM how to fix this.  But could we dedupe/reuse the switchableTpe/SwitchablePattern/etc that's in RegularSwitchMaker?  We wouldn't want the logic to become askew.

Reply:
When I saw that the pieces are disjoint, I realized also that they are logically disjoint: that is, my first attempt was to use a `RSM` to judge switchability, though it would need a tweak and was not accessible. But it was cleaner just to squint and sketch a few lines of code. This code doesn't care whether the optimization is actually applied; for example, the judgment about how many cases are required happens in two places; and `SwitchMaker` is generally subtle and buried in the cake somewhere. I think in this case, local duplication is cleaner.

Have we punned on cake and mix-in as in "cake mix", the kind in a box from the grocery? Or was that an ingredient of the foundational pun of the cake pattern?

Reply:
I think the pun was missed initially, as it went mixins -> layers -> cake.

Seems like a bit of code to have in common, so if it's too hard, let's at least have a common reference to one another, for future spelunkers.